src/bootstrap/builder.rs

```diff
@@ -1111,6 +1111,15 @@ impl<'a> Builder<'a> {
             );
         }
 
+        // If Control Flow Guard is enabled, pass the `control_flow_guard=checks` flag to rustc.
+        // This is not needed for stage 0.
```

Comment:
Is this because we don't yet support it on the bootstrap compiler, or for some other reason? Expanding on the comment would be good either way.

Reply:
I guess this is related to the point above: I'm assuming the stage 0 artifacts won't be used to build anything other than the stage 1 compiler, so wouldn't need CFG protection (and also that it's not yet supported by the bootstrap compiler).

Does this make sense? If so, I'll update the comment.

Reply:
If I'm following correctly that this is a security mitigation, then I would expect that we want it for all code, including the compiler? Why is the compiler special?

Codegen isn't going to be linked into downstream crates though so that can be dropped if we go with code linked into all Rust libraries.

Reply:
Yes, this is a security mitigation, but at present it's only available on Windows. I agree that ideally we should also enable this for the compiler and tools, but I think this should be a separate option from enabling it for the standard library. For example, if a developer is using a non-Windows platform, they might still want to build a version of the standard library with CFG enabled (but not enable it for the compiler and tools).

I've updated the comment to reflect this, and I've also added documentation about how to build the standard library using `-Z build-std`.